core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1238,6 +1239,18 @@ package object config {
         s"The value must be in allowed range [1,048,576, ${MAX_BUFFER_SIZE_BYTES}].")
       .createWithDefault(1024 * 1024)
 
+  private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_RATIO =
+    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.size.ratio")
+      .doc("The multiplication ratio is the parameter that controls the initial read buffer " +
+        "size. The multiplication ratio value range is from 1 to 1024. This parameter changes "  +
+        "the initial read buffer size in 1KB increments. It will result in the initial buffer " +
+        "size in the range from 1KB to 1MB. The read buffer size is dynamically adjusted " +
+        "afterward based on data length read from the spilled file.")
+      .intConf
+      .checkValue(v => 1 <= v && v <= DEFAULT_BUFFER_SIZE_RATIO,
+        s"The value must be in allowed range [1, ${DEFAULT_BUFFER_SIZE_RATIO}].")
+      .createWithDefault(DEFAULT_BUFFER_SIZE_RATIO)
```

Comment:
Could you move this config into `SQLConf`?

Reply:
Just need clarification. SQLConf is member of catalist package/project. I looked all classes involved in this change and I do not see usage of SQLConf. I would expect that new parameter UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_RATIO is together with existing package$.MODULE$.UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE parameter  in package.scala. I just want to be sure that we want UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_RATIO to be in SQLConf. Could you please confirm?

Reply:
Ah, yes. My last comment's incorrect. Your update looks okay.